src/features/slackapi/slackapi-utils.ts

```diff
@@ -0,0 +1,111 @@
+import { FormattedTimebankData } from "@functions/sendslack/schema";
+import { LogLevel, WebClient } from "@slack/web-api";
+import { Member } from "@slack/web-api/dist/response/UsersListResponse";
+import dotenv from "dotenv";
+import { DateTime, Duration } from "luxon";
+
+dotenv.config({ path: __dirname + "/../../../../../.env" });
+
+/**
+ * Namespace for Slack API utilities
+ */
+namespace SlackApiUtilities {
+
+  const client = new WebClient(process.env.metatavu_bot_token, {
+    logLevel: LogLevel.DEBUG,
+  });
+
+  /**
+   * Get list of slack users
+   *
+   * @returns Promise of slack user data
+   */
+  export const getSlackUserData = async (): Promise<Member[]> => {
+    let userData: Member[] = [];
+
+    /**
+     * Put Slack Users from getUsersList into userData
+     *
+     * @param usersArray
+     */
+    const saveUsers = (usersArray: Member[]) => {
+      userData = usersArray.map((user) => {
+        return { name: user.real_name, id: user.id };
+      });
+    };
+
+    /**
+     * Request list of Slack Users
+     */
+    const getUsersList = async () => {
+      try {
+        const result = await client.users.list();
+        saveUsers(result.members);
+      } catch (error) {
+        console.error("Error while loading slack users list");
+        Promise.reject(error);
+      }
+    };
+    await getUsersList();
+    return userData;
+  };
```

Comment:
This can be simplified into:
```suggestion
  /**
   * Get list of slack users
   *
   * @returns Promise of slack user data
   */
  export const getSlackUsers = async (): Promise<Member[]> => {
    try {
      const result = await client.users.list();
      return result.members;
    } catch (error) {
      console.error("Error while loading slack users list");
      Promise.reject(error);
    }
  };
```

Reply:
understood, thanks